R/fullVALIDATIONREMIND.R

```diff
@@ -13,46 +13,57 @@
-    # read mapping file
-    new <- colnames(toolGetMapping(mappings[r], type = "regional"))
-    # ignore "X" and "CountryCode"
-    new <- new[!new %in% c("X", "CountryCode")]
-    # check if columns of current mapping exist in any mapping read before
-    if (any(new %in% rel)) warning("The following column(s) from ", mappings[r],
-                                   " exist in another mapping an will be ignored: ",
-                                   paste(new[new %in% rel], collapse = ", "))
-    # append new columns removing duplicates
-    rel <- unique(c(rel, new))
-  }
-
-  #-------------- use columns from all regionmappings for aggregation ---------------------------------
-
-  columnsForAggregation <- gsub("RegionCode", "region", paste0(c(rel, "global"), collapse = "+"))
+  rel <- 'global'   # always compute global aggregate
+  for (mapping in c(getConfig("regionmapping"), getConfig("extramappings"))) {
+    columns <- setdiff(colnames(toolGetMapping(mapping, 'regional')),
+                       c('X', 'CountryCode'))
 
-  #-------------- historical data ---------------------------------------------------------------------
+    if (any(columns %in% rel))
+      warning("The following column(s) from ", mapping,
+              " exist in another mapping an will be ignored: ",
+              paste(columns[columns %in% rel], collapse = ", "))
 
-  valfile <- "historical.mif"
+    rel <- unique(c(rel, columns))
+  }
 
-  calcOutput("Historical", round = 5,  file = valfile, aggregate = columnsForAggregation,
-             append = FALSE, na_warning = FALSE, try = TRUE)
+  columnsForAggregation <- gsub("RegionCode", "region",
+                                paste(rel, collapse = "+"))
 
-  calcOutput(type = "IEA_ETP", aggregate = "global", file = valfile,
-             append = TRUE, na_warning = FALSE, try = TRUE, isValidation = TRUE)
+  # default arguments ----
+  # default arguments used for calling all subsequent `calc` functions
+  default_arguments <- list(
+    file = 'historical.mif',
+    append = TRUE,
+    warnNA = FALSE,
+    aggregate = columnsForAggregation,
+    try = TRUE)
 
-  calcOutput(type = "IEA_ETP", aggregate = "region", file = valfile,
-             append = TRUE, na_warning = FALSE, try = TRUE, isValidation = TRUE)
+  # list `calc` functions to call ----
+  # list of all `calc` functions to call, as well as their specific arguments
+  # arguments can be empty (`list()`), and can overwrite default arguments if
+  # needed
+  historical_data_calls <- list(
+    'Historical'   = list(round = 5),
+    'IEA_ETP'      = list(isValidation = TRUE),
+    'IEA_WEO_2021' = list(isValidataion = TRUE),
+    'UNIDO'        = list(subtype = 'INDSTAT2') # industry value added
+    )
 
-  calcOutput(type = "IEA_WEO_2021", aggregate = "global", file = valfile,
-             append = TRUE, na_warning = FALSE, try = TRUE, isValidation = TRUE)
+  # call all listed functions ----
+  for (i in seq_along(historical_data_calls)) {
```

Comment:
I am against it. Removing redundancy makes this part much harder to understand for the 95% of people who don't have an IT-Masters and are working with this. As this is an entry point to understanding where validation data is coming from, being accessible to everyone has priority here.